go-controller/pkg/ovn/pods.go

```diff
@@ -139,8 +139,9 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod) (err error) {
 	klog.Infof("Deleting pod: %s", podDesc)
 
 	logicalPort := util.GetLogicalPortName(pod.Namespace, pod.Name)
-	portInfo, err := oc.logicalPortCache.get(logicalPort)
-	if err != nil {
+	// using retryPods instead of logicalPortCache ensures we are deleting the right port
+	portInfo := oc.retryPods[logicalPort].logicalPortCacheEntry
+	if portInfo == nil {
 		klog.Errorf(err.Error())
 		// If ovnkube-master restarts, it is also possible the Pod's logical switch port
 		// is not re-added into the cache. Delete logical switch port anyway.
```

Comment:
so in this part, we will still have that brief problem, where we end up accidentally deleting the lsp of a good new pod :/

Reply:
😮  nothing being introduced by this PR, or is it?
if ls is gone, we can safely assume lsp is also gone. Should we account for those cases and keep going towards `oc.lsManager.ReleaseIPs(` ?